client/me/account/toggle-new-hosting-dashboard.tsx

```diff
@@ -0,0 +1,43 @@
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch, useSelector } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { savePreference } from 'calypso/state/preferences/actions';
+import {
+	getPreference,
+	isSavingPreference,
+	isFetchingPreferences,
+} from 'calypso/state/preferences/selectors';
+
+export default function ToggleNewHostingDashboard() {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const enableNewHostingDashboard = useSelector( ( state ) =>
+		getPreference( state, 'enable-hosting-dashboard-v2' )
```

Comment:
I'm using a Calypso _preference_ here, rather than a user setting. For a comparison, the "community translator" preference is stored as a _setting_. I believe the difference is that the setting is stored as a user attribute and is probably easier to access in PHP.

I'm not sure whether there are any advantages of using a _setting_ over just using the Calypso preference feature.